lib/rucio/core/replica.py

```diff
@@ -1830,17 +1830,22 @@ def _expected_pfns(lfns, rse_settings, scheme, operation='write', domain='wan',
                 # Check that the pfns match to the expected pfns
                 lfns = [{'scope': i['scope'].external, 'name': i['name']} for i in files if i['pfn'].startswith(scheme)]
                 pfns[scheme] = clean_pfns(pfns[scheme])
+                expected_pfns = set()
 
                 for protocol_attr in rsemgr.get_protocols_ordered(rse_settings=rse_settings, operation='write', scheme=scheme, domain='wan'):
-                    pfns[scheme] = list(set(pfns[scheme]) - set(_expected_pfns(lfns, rse_settings, scheme, operation='write', domain='wan', protocol_attr=protocol_attr)))
+                    expected_pfn = set(_expected_pfns(lfns, rse_settings, scheme, operation='write', domain='wan', protocol_attr=protocol_attr))
+                    expected_pfns = expected_pfns.union(expected_pfn)
```

Comment:
Imo it would be cleaner to do

```suggestion
                    expected_pfns.update(_expected_pfns(lfns, rse_settings, scheme, operation='write', domain='wan', protocol_attr=protocol_attr)
```

Just to skip having to do the `list`->`set` conversion first, and then the `union`. `set.update()` likely does this in the background anyway, but given that we're iterating in a loop, `update` feels more appropriate